llarp/router/outbound_session_maker.cpp

```diff
@@ -0,0 +1,293 @@
+#include <router/outbound_session_maker.hpp>
+
+#include <llarp/link/server.hpp>
+#include <util/logic.hpp>
+#include <router_contact.hpp>
+#include <nodedb.hpp>
+#include <router/i_rc_lookup_handler.hpp>
+#include <link/i_link_manager.hpp>
+
+namespace llarp
+{
+
+  struct PendingSession
+  {
+    //TODO: add session establish status metadata, e.g. num retries
+
+    const RouterContact rc;
+    LinkLayer_ptr link;
+
+    size_t attemptCount = 0;
+
+    PendingSession(const RouterContact &rc,
+                   LinkLayer_ptr link)
+        : rc(rc)
+        , link(link)
+    {
+    }
+
+  };
+
+  bool
+  OutboundSessionMaker::OnSessionEstablished(ILinkSession *session)
+  {
+    //TODO: do we want to keep it
+
+    const auto &router = session->GetPubKey();
+
+    LogInfo("session with ", router, " established");
+
+    if (not _rcLookup->RemoteIsAllowed(router))
+    {
+      FinalizeRequest(router, SessionResult::InvalidRouter);
+    }
+
+    auto func = std::bind(&OutboundSessionMaker::VerifyRC,
+                          this,
+                          session->GetRemoteRC());
+    _threadpool->addJob(func);
```

Comment:
fwiw, I think naming `func` detracts from the semantics of the call.

```c++
_threadpool->addJob(std::bind(&OutboundSessionMaker::VerifyRC, this, session->GetRemoteRC()));
```

Reply:
If it weren't the line just before I'd agree.

Reply:
Not changed.